packages/ui/src/composables/useWeb3.ts

```diff
@@ -230,17 +240,48 @@ export function useWeb3() {
       if (erc20Contract) {
         const erc20PlayerInfo: RedeemPlayerInfo | null =
           await getRedeemPlayerInfo()
+        const walletInfo = await getWalletInfo({
+          walletAddress: gameStore.provider.address,
+        })
         const erc20Info: ERC20ContractInfo | null = await getERC20ContractInfo()
-        if (erc20PlayerInfo?.playerAddress === ZERO_ADDRESS) {
-          gameStore.setGameOverStatus(GameOverStatus.AllowRedeem)
-        } else {
-          if (erc20Info?.status == ERC20Status.Acquired) {
-            gameStore.setGameOverStatus(GameOverStatus.AllowWithdraw)
-          } else if (erc20Info?.status == ERC20Status.Auctioning) {
-            gameStore.setGameOverStatus(GameOverStatus.AllowSale)
+        const transactionConfirmed = computed(
+          () =>
+            localStore.txInfo?.txHash &&
+            localStore.txInfo?.blockNumber &&
+            localStore.txInfo?.txConfirmation
+        )
+        // No transaction in progress
+        if (transactionConfirmed.value || !localStore.txInfo?.txHash) {
+          // There is player info and player didn't claim ownership
+          if (erc20PlayerInfo?.playerAddress === ZERO_ADDRESS) {
+            gameStore.setGameOverStatus(GameOverStatus.AllowRedeem)
           } else {
-            gameStore.setGameOverStatus(GameOverStatus.AwaitSale)
-            //GET REDEEM INFO
+            // No player info or the player already claimed ownership
+            if (erc20Info?.status == ERC20Status.Acquired) {
+              // The NFT is acquired
+              if (walletInfo?.wpxBalance > 0) {
+                // Players with $WPX balance can swap to $ETH
```

Comment:
(not ony players, non-players holders as well) => // Holders of $WPX can swap to $ETH